sympy/core/relational.py

```diff
@@ -494,11 +495,34 @@ def _eval_simplify(self, **kwargs):
                             del nzm[0]
                             newexpr = Add(*[i * j for i, j in nzm])
                             r = r.func(lhsterm, -newexpr)
-
                     else:
                         r = r.func(constant, S.Zero)
                 except ValueError:
-                    pass
+                    try:
+                        from sympy.polys.polyerrors import PolynomialError
+                        from sympy.polys.polytools import gcd, Poly, poly
+                        p = poly(dif, free[0])
+                        c = p.all_coeffs()
+                        constant = c[-1]
+                        c[-1] = 0
```

Comment:
This is copying a large block of code from above duplicating many things that should not be duplicated and also that should not be repeated at runtime.